app/bankaccountverification/web/personal/PersonalVerificationRequest.scala

```diff
@@ -23,15 +23,18 @@ import bankaccountverification.web.Forms._
 import bankaccountverification.web.Implicits.SanitizedString
 import play.api.data.Form
 import play.api.data.Forms._
-import play.api.libs.json.Json
+import play.api.libs.json.{Json, OWrites, Reads, Writes}
 
-case class PersonalVerificationRequest private (accountName: String, sortCode: String, accountNumber: String, rollNumber: Option[String] = None)
+case class PersonalVerificationRequest private(accountName: String, sortCode: String, accountNumber: String, rollNumber: Option[String] = None)
 
 object PersonalVerificationRequest {
   def apply(accountName: String, sortCode: String, accountNumber: String,
             rollNumber: Option[String] = None): PersonalVerificationRequest = {
     val cleanSortCode = sortCode.stripSpacesAndDashes()
-    val cleanAccountNumber = accountNumber.stripSpacesAndDashes().leftPadToLength(8, '0')
+    val cleanAccountNumber = {
+      val nsAccountNumber = accountNumber.stripSpacesAndDashes()
+      if (nsAccountNumber.isEmpty) "" else nsAccountNumber.leftPadToLength(8, '0')
```

Comment:
why are we using "" instead of None?

Reply:
This is because we have the pre-populated object being passed in. We could change this to a None but not sure that would anything and also make the handling code more involved.